src/agent/handlers/nodesHolePunchSignal.ts

```diff
@@ -0,0 +1,54 @@
+import type { AgentRPCRequestParams, AgentRPCResponseResult } from '../types';
+import type NodeConnectionManager from '../../nodes/NodeConnectionManager';
+import type { Host, Port } from '../../network/types';
+import type { NodeAddress } from '../../nodes/types';
+import type { HolePunchSignalMessage } from './types';
+import type { JSONValue } from '../../types';
+import { never } from '../../utils';
+import * as agentErrors from '../errors';
+import * as agentUtils from '../utils';
+import { UnaryHandler } from '../../rpc/handlers';
+import * as validationUtils from '../../validation/utils';
+
+class NodesHolePunchSignalHandler extends UnaryHandler<
```

Comment:
It is not clear what this separation is doing?

You need 2 handlers right?

1. One for receiving a signalling request.
2. One for receiving the hole punching request.

The reason why these calls were the same was so that it would eventually lead to decentralised signalling.

I'm not sure how these 2 calls make sense...

The naming of these 2 is problematic, because we should name them as RPC calls that are called by the origin node.

You could say:

* A -nodesConnect -> S
* S -nodesConnect -> B
* A -> B

It's supposed to be the same call, so that this may be possible:

* A -nodesConnect -> S1
* S1 -nodesConnect -> S2
* S2 -nodesConnect -> B
* A -*> B

Furthermore I don't see any authentication being done here. I don't see how this achieves #148.

Reply:
Authentication is being addressed now by the latest changes.

The names have also be updated.

As for how it's designed now. We will only every send a request to an single relaying node. There will be no multi-hop requests. It's not needed now and won't be needed after #365 is done. SO the steps are clearly request -> relay -> punch. It simplifies the logic.